Documentation/Diagnostics/PH2145.md

```diff
@@ -0,0 +1,61 @@
+# PH2145: Prefer interpolated strings over string.Format
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2145 |
+| Category  | [Readability](../Readability.md) |
+| Analyzer | [PreferInterpolatedStringAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/main/Philips.CodeAnalysis.MaintainabilityAnalyzers/Readability/PreferInterpolatedStringAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+Prefer interpolated strings over `string.Format` calls with simple placeholders for better readability and reduced error-proneness.
+
+## How to solve
+
+Replace `string.Format` calls that use simple placeholders like `{0}`, `{1}`, etc. with interpolated strings.
+
+## Example
+
+Code that triggers a diagnostic:
+``` cs
+class BadExample
+{
+    public void BadMethod()
+    {
+        int num = 42;
+        string name = "World";
+        string str = string.Format("Hello {0}, this is number {1}", name, num);
+    }
+}
+
+```
+
+And the replacement code:
+``` cs
+class GoodExample
+{
+    public void GoodMethod()
+    {
+        int num = 42;
+        string name = "World";
+        string str = $"Hello {name}, this is number {num}";
+    }
+}
+
+```
+
+## What it ignores
+
+The analyzer is conservative and only suggests conversion in safe cases:
+
+- **No placeholders**: `string.Format("Simple string")` - no benefit from conversion
```

Comment:
Regarding "No placeholders", this feels like an easy addition. It could be the same diagnostic with a different error message, e.g., "unnecessary call to string.Format".